src/main/java/com/mathworks/ci/TestResultsViewAction.java

```diff
@@ -0,0 +1,265 @@
+        FilePath fl = new FilePath(new File(build.getRootDir().getAbsolutePath() + File.separator + MatlabBuilderConstants.TEST_RESULTS_VIEW_ARTIFACT + this.actionID + ".json"));
+        try (InputStreamReader reader = new InputStreamReader(Files.newInputStream(Paths.get(fl.toURI())), StandardCharsets.UTF_8)) {
+            this.totalCount = 0;
+            this.passedCount = 0;
+            this.failedCount = 0;
+            this.incompleteCount = 0;
+            this.notRunCount = totalCount;
+
+            JSONArray testArtifact = (JSONArray) new JSONParser().parse(reader);
+            Iterator<JSONArray> testArtifactIterator = testArtifact.iterator();
+
+            while(testArtifactIterator.hasNext()){
+                Object jsonTestSessionResults = testArtifactIterator.next();
+
+                List<MatlabTestFile> testSessionResults = new ArrayList<>();
+                Map<String, MatlabTestFile> map = new HashMap<>();
+
+                if(jsonTestSessionResults instanceof JSONArray){
+                    JSONArray jsonTestSessionResultsArray = (JSONArray) jsonTestSessionResults;
+                    Iterator<JSONObject> testSessionResultsIterator = jsonTestSessionResultsArray.iterator();
+
+                    while(testSessionResultsIterator.hasNext()){
+                        JSONObject jsonTestCase = testSessionResultsIterator.next();
+                        getTestSessionResults(testSessionResults, jsonTestCase, map);
+                    }
+                }
+                else if(jsonTestSessionResults instanceof JSONObject) {
+                    JSONObject jsonTestCase = (JSONObject) jsonTestSessionResults;
+                    getTestSessionResults(testSessionResults, jsonTestCase, map);
+                }
+
+                testResults.add(testSessionResults);
+            }
+        }
+        catch (Exception e) {
+            throw new IOException(e.getLocalizedMessage());
+        }
+
+        return testResults;
+    }
+
+    private void getTestSessionResults(List<MatlabTestFile> testSessionResults, JSONObject jsonTestCase, Map<String, MatlabTestFile> map) throws IOException, InterruptedException {
+        FilePath baseFolder = new FilePath(new File(jsonTestCase.get("BaseFolder").toString()));
+        JSONObject matlabTestCaseResult = (JSONObject) jsonTestCase.get("TestResult");
+
+        // Not OS dependent
+        String[] testNameSplit = matlabTestCaseResult.get("Name").toString().split("/");
+        String matlabTestFileName = testNameSplit[0];
+        String matlabTestCaseName = testNameSplit[1];
+
+        // Check if test's file was known or not
+        MatlabTestFile matlabTestFile = map.get(baseFolder + File.separator + matlabTestFileName);
+        if(matlabTestFile == null) {
+            matlabTestFile = new MatlabTestFile();
+            matlabTestFile.setName(matlabTestFileName);
+
+            map.put(baseFolder + File.separator + matlabTestFileName, matlabTestFile);

```

Comment:
Does `File.separator` work if the Jenkins server is running on Windows but the job is run on a Linux node? I was seeing some weirdness with that case with the mpm implementation, but I'm not sure if all of this code runs on the server or if part is running on the node as well. We might want to make sure we bash that case.

Reply:
I remember Sam had mentioned it earlier, since all the action code would only be running on the agent we shouldn't face issues there. But it would be good to bash this scenario, thanks for reminding!